Fix wrong line number in private key detector (#4485)

kmgv opened the pull request:
### Description:

Private key detector findings report wrong line number when private key literal doesn't end with new line character.

After private is matched using regexp is goes through Normalize function and normalized result is used in result.Raw and then used in engine.FragmentLineOffset which looks for line of code.

Normalization step is crucial as ssh.ParseRawPrivateKey is quite strict about format of accepted key and this step can sieve false posivites as it can verify it private key is legit or just matches permisive regexp.

Normalize always adds newline char at end of string (as needed for validation) but such string, with new line at the end is then used for looking for LOC. If source chunk didn't have new line char right after private key engine will report default LOC.

This fix changes Result.Raw for private key detector to use raw match from regexp and not normalized string. This way engine can calculate correct LOC for such finding.

Related issue #4485 

### Checklist:
* [X] Tests passing (`make test-community`)?
* [ ] Lint passing (`make lint` this requires [golangci-lint](https://golangci-lint.run/welcome/install/#local-installation))? - it doesn't even pass for main branch locally

kmgv committed:
Fix wrong line number in private key detector (#4485)

Private key detector findings report wrong line number when private key
literal doesn't end with new line character.

After private is matched using regexp is goes through Normalize
function and normalized result is used in result.Raw and then used in
engine.FragmentLineOffset which looks for line of code.

Normalization step is crucial as ssh.ParseRawPrivateKey is quite strict
about format of accepted key and this step can sieve false posivites as
it can verify it private key is legit or just matches permisive regexp.

Normalize always adds newline char at end of string (as needed for
validation) but such string, with new line at the end is then used for
looking for LOC. If source chunk didn't have new line char right after
private key engine will report default LOC.

This fix changes Result.Raw for private key detector to use raw match
from regexp and not normalized string. This way engine can calculate
correct LOC for such finding.

kashifkhan0771 commented on pkg/detectors/privatekey/privatekey.go:
```diff
@@ -66,7 +66,7 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 		s1 := detectors.Result{
 			DetectorType: detectorspb.DetectorType_PrivateKey,
-			Raw:          []byte(token),
+			Raw:          []byte(match),
```
We don't recommend updating the `Raw` field value, as it's used as an identifier and changing it can interfere with existing findings. Your solution is correct, but instead, you can use:

```go
s1.SetPrimarySecretValue(match)
```

This will instruct the engine to use `match` to determine the line number.

kashifkhan0771 committed:
Merge branch 'main' into bugfix/private-key-loc

shahzadhaider1 committed:
Merge branch 'main' into bugfix/private-key-loc

kmgv committed:
Fix wrong line number in private key detector - use primary primarySecret (#4485)

Revert previous changes that changed result.Raw in primary key detector
as it  can interfere with existing finding. Use SetPrimarySecretValue(match)
instead

kashifkhan0771 committed:
Merge branch 'main' into bugfix/private-key-loc

rosecodym reviewed (approved):
Thanks for this! I'm mostly deferring to @kashifkhan0771 :)
